### PR TITLE
Add e2e tests to account linking

### PR DESCRIPTION
**Pull request by tung2744**
ref DEV-1229
ref https://github.com/authgear/authgear-server/issues/4143
requires https://github.com/authgear/authgear-server/pull/4166

### CODE REVIEW

**Comment by louischan-oursky** [comment]
Please rebase.

File: e2e/tests/account_linking/email_users.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+  "identifier": "email",
+  "records": [
+    {
+      "email": "mock@authgear.com",
```
Comment (louischan-oursky) [inline e2e/tests/account_linking/email_users.json:5]: Should be example.com See https://github.com/authgear/authgear-server/pull/4128

**Comment by louischan-oursky** [comment]
@tung2744 I realized that the tests failed actually. But `make run` does not exit >0 https://github.com/authgear/authgear-server/actions/runs/8934707885/job/24542012277?pr=4192

**Comment by tung2744** [comment]
Fixed the e2e tests and also modified the script to return correct exit status, thanks!

**Pull request merged**